### PR TITLE
Enhance - `azurerm_redis_cache` - Recreate Redis cache when downgrade the SKU

### LINKED ISSUE #17740: azurerm_redis_cache fails to downgrade SKU
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

2.99.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "this" {
  name     = "redis-rg"
  location = "North Europe"
}

resource "azurerm_redis_cache" "this" {
  name                = "my-redis"
  location            = azurerm_resource_group.this.location
  resource_group_name = azurerm_resource_group.this.name
  capacity            = 0
  family              = "C"
  sku_name            = "Standard"
  enable_non_ssl_port = false
  minimum_tls_version = "1.2"

  redis_configuration {
  }
}
```

### Debug Output/Panic Output

```shell
│ Error: updating Redis Cache "my-redis" (Resource Group "redis-rg"): redis.Client#Update: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Invalid update request: properties.sku.name cannot be updated to Basic.\r\nRequestID=xxx"
│ 
│   with module.redis.azurerm_redis_cache.redis,
│   on .terraform/modules/redis/main.tf line 98, in resource "azurerm_redis_cache" "this":
│   98: resource "azurerm_redis_cache" "this" {
```

### Expected Behaviour

The provider should detect `sku` downgrade with following plan and attempt to recreate the Redis cache instead of doing an in-place update.

```
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.redis.azurerm_redis_cache.this will be updated in-place
  ~ resource "azurerm_redis_cache" "this" {
        id                            = "xxx"
        name                          = "my-redis"
      ~ sku_name                      = "Standard" -> "Basic"
        # (20 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Actual Behaviour

Terraform apply fails

### Steps to Reproduce

1. terraform apply

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by xuzhang3**
Fixes #17740 
Downgrade SKU is not supported by service, custom the `forceNew` to true when users try to downgrade the SKU.

### CODE REVIEW

**Comment by xuzhang3** [comment]
```log
=== RUN   TestAccRedisCache_basic
=== PAUSE TestAccRedisCache_basic
=== RUN   TestAccRedisCache_withoutSSL
=== PAUSE TestAccRedisCache_withoutSSL
=== RUN   TestAccRedisCache_requiresImport
=== PAUSE TestAccRedisCache_requiresImport
=== RUN   TestAccRedisCache_standard
=== PAUSE TestAccRedisCache_standard
=== RUN   TestAccRedisCache_premium
=== PAUSE TestAccRedisCache_premium
=== RUN   TestAccRedisCache_premiumSharded
=== PAUSE TestAccRedisCache_premiumSharded
=== RUN   TestAccRedisCache_premiumShardedScaling
=== PAUSE TestAccRedisCache_premiumShardedScaling
=== RUN   TestAccRedisCache_BackupDisabled
=== PAUSE TestAccRedisCache_BackupDisabled
=== RUN   TestAccRedisCache_BackupEnabled
=== PAUSE TestAccRedisCache_BackupEnabled
=== RUN   TestAccRedisCache_BackupEnabledDisabled
=== PAUSE TestAccRedisCache_BackupEnabledDisabled
=== RUN   TestAccRedisCache_AOFBackupEnabled
=== PAUSE TestAccRedisCache_AOFBackupEnabled
=== RUN   TestAccRedisCache_AOFBackupEnabledDisabled
=== PAUSE TestAccRedisCache_AOFBackupEnabledDisabled
=== RUN   TestAccRedisCache_PatchSchedule
=== PAUSE TestAccRedisCache_PatchSchedule
=== RUN   TestAccRedisCache_PatchScheduleUpdated
=== PAUSE TestAccRedisCache_PatchScheduleUpdated
=== RUN   TestAccRedisCache_PublicNetworkAccessDisabledEnabled
=== PAUSE TestAccRedisCache_PublicNetworkAccessDisabledEnabled
=== RUN   TestAccRedisCache_InternalSubnet
=== PAUSE TestAccRedisCache_InternalSubnet
=== RUN   TestAccRedisCache_InternalSubnetStaticIP
=== PAUSE TestAccRedisCache_InternalSubnetStaticIP
=== RUN   TestAccRedisCache_InternalSubnet_withZone
=== PAUSE TestAccRedisCache_InternalSubnet_withZone
=== RUN   TestAccRedisCache_SubscribeAllEvents
=== PAUSE TestAccRedisCache_SubscribeAllEvents
=== RUN   TestAccRedisCache_WithoutAuth
=== PAUSE TestAccRedisCache_WithoutAuth
=== RUN   TestAccRedisCache_ReplicasPerMaster
=== PAUSE TestAccRedisCache_ReplicasPerMaster
=== RUN   TestAccRedisCache_ReplicasPerPrimary
=== PAUSE TestAccRedisCache_ReplicasPerPrimary
=== RUN   TestAccRedisCache_RedisVersion
=== PAUSE TestAccRedisCache_RedisVersion
=== RUN   TestAccRedisCache_RedisVersionUpgrade
=== PAUSE TestAccRedisCache_RedisVersionUpgrade
=== RUN   TestAccRedisCache_TenantSettings
=== PAUSE TestAccRedisCache_TenantSettings
=== RUN   TestAccRedisCache_redisConfiguration
=== PAUSE TestAccRedisCache_redisConfiguration
=== RUN   TestAccRedisCache_identity
=== PAUSE TestAccRedisCache_identity
=== CONT  TestAccRedisCache_basic
=== CONT  TestAccRedisCache_PublicNetworkAccessDisabledEnabled
=== CONT  TestAccRedisCache_BackupDisabled
=== CONT  TestAccRedisCache_PatchScheduleUpdated
=== CONT  TestAccRedisCache_premiumShardedScaling
=== CONT  TestAccRedisCache_premiumSharded
=== CONT  TestAccRedisCache_premium
=== CONT  TestAccRedisCache_standard
=== CONT  TestAccRedisCache_requiresImport
--- PASS: TestAccRedisCache_BackupDisabled (1149.04s)
--- PASS: TestAccRedisCache_premium (1245.04s)
=== CONT  TestAccRedisCache_withoutSSL
--- PASS: TestAccRedisCache_PatchScheduleUpdated (1343.71s)
=== CONT  TestAccRedisCache_AOFBackupEnabled
--- PASS: TestAccRedisCache_premiumSharded (1432.17s)
=== CONT  TestAccRedisCache_PatchSchedule
--- PASS: TestAccRedisCache_basic (2025.32s)
=== CONT  TestAccRedisCache_AOFBackupEnabledDisabled
--- PASS: TestAccRedisCache_standard (2117.62s)
=== CONT  TestAccRedisCache_BackupEnabledDisabled
--- PASS: TestAccRedisCache_PublicNetworkAccessDisabledEnabled (2299.37s)
=== CONT  TestAccRedisCache_SubscribeAllEvents
--- PASS: TestAccRedisCache_AOFBackupEnabled (1308.22s)
=== CONT  TestAccRedisCache_ReplicasPerMaster
--- PASS: TestAccRedisCache_premiumShardedScaling (2680.79s)
=== CONT  TestAccRedisCache_WithoutAuth
--- PASS: TestAccRedisCache_PatchSchedule (1269.11s)
=== CONT  TestAccRedisCache_InternalSubnetStaticIP
--- PASS: TestAccRedisCache_requiresImport (1934.32s)
=== CONT  TestAccRedisCache_InternalSubnet_withZone
--- PASS: TestAccRedisCache_withoutSSL (1838.56s)
=== CONT  TestAccRedisCache_BackupEnabled
--- PASS: TestAccRedisCache_BackupEnabledDisabled (1533.36s)
=== CONT  TestAccRedisCache_TenantSettings
--- PASS: TestAccRedisCache_SubscribeAllEvents (1367.99s)
=== CONT  TestAccRedisCache_identity
--- PASS: TestAccRedisCache_ReplicasPerMaster (1332.92s)
=== CONT  TestAccRedisCache_redisConfiguration
--- PASS: TestAccRedisCache_AOFBackupEnabledDisabled (2066.99s)
=== CONT  TestAccRedisCache_InternalSubnet
--- PASS: TestAccRedisCache_BackupEnabled (1523.32s)
=== CONT  TestAccRedisCache_RedisVersionUpgrade
--- PASS: TestAccRedisCache_TenantSettings (1137.35s)
=== CONT  TestAccRedisCache_RedisVersion
--- PASS: TestAccRedisCache_redisConfiguration (1758.95s)
=== CONT  TestAccRedisCache_ReplicasPerPrimary
--- PASS: TestAccRedisCache_identity (2231.71s)
--- PASS: TestAccRedisCache_RedisVersion (1216.86s)
--- PASS: TestAccRedisCache_RedisVersionUpgrade (1717.38s)
--- PASS: TestAccRedisCache_ReplicasPerPrimary (1271.63s)
=== CONT  TestAccRedisCache_WithoutAuth
--- PASS: TestAccRedisCache_WithoutAuth (6049.41s)
=== CONT  TestAccRedisCache_InternalSubnetStaticIP
--- PASS: TestAccRedisCache_InternalSubnetStaticIP (6038.11s)
=== CONT  TestAccRedisCache_InternalSubnet_withZone
--- PASS: TestAccRedisCache_InternalSubnet_withZone (5671.93s)
=== CONT  TestAccRedisCache_InternalSubnet
--- PASS: TestAccRedisCache_InternalSubnet (4674.85s)
PASS
PASS    github.com/hashicorp/terraform-provider-azurerm/internal/services/redis 8783.416s
PASS
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @xuzhang3, just one suggestion - it might be good to include a note in the docs reflecting this?

**Comment by xuzhang3** [comment]
@catriona-m doc updated

**Comment by xuzhang3** [comment]
@catriona-m  doc updated.

File: website/docs/r/redis_cache.html.markdown [hunk 1]
```diff
@@ -55,6 +55,8 @@ The following arguments are supported:
 
 * `sku_name` - (Required) The SKU of Redis to use. Possible values are `Basic`, `Standard` and `Premium`.
 
+~> **Note** Downgrade the SKU forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/redis_cache.html.markdown:58]:
```suggestion
~> **Note** Downgrading the SKU will force a new resource to be created.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Can we add a test that downgrades the SKU?

**Comment by xuzhang3** [comment]
@katbyte  thanks for review, test case added. 

```log
=== RUN   TestAccRedisCache_basic
=== PAUSE TestAccRedisCache_basic
=== RUN   TestAccRedisCache_withoutSSL
=== PAUSE TestAccRedisCache_withoutSSL
=== RUN   TestAccRedisCache_requiresImport
=== PAUSE TestAccRedisCache_requiresImport
=== RUN   TestAccRedisCache_standard
=== PAUSE TestAccRedisCache_standard
=== RUN   TestAccRedisCache_premium
=== PAUSE TestAccRedisCache_premium
=== RUN   TestAccRedisCache_premiumSharded
=== PAUSE TestAccRedisCache_premiumSharded
=== RUN   TestAccRedisCache_premiumShardedScaling
=== PAUSE TestAccRedisCache_premiumShardedScaling
=== RUN   TestAccRedisCache_BackupDisabled
=== PAUSE TestAccRedisCache_BackupDisabled
=== RUN   TestAccRedisCache_BackupEnabled
=== PAUSE TestAccRedisCache_BackupEnabled
=== RUN   TestAccRedisCache_BackupEnabledDisabled
=== PAUSE TestAccRedisCache_RedisVersion
=== RUN   TestAccRedisCache_RedisVersionUpgrade
=== PAUSE TestAccRedisCache_RedisVersionUpgrade
=== RUN   TestAccRedisCache_TenantSettings
=== PAUSE TestAccRedisCache_TenantSettings
=== RUN   TestAccRedisCache_redisConfiguration
=== PAUSE TestAccRedisCache_redisConfiguration
=== RUN   TestAccRedisCache_identity
=== PAUSE TestAccRedisCache_identity
=== RUN   TestAccRedisCache_SkuDowngrade
=== PAUSE TestAccRedisCache_SkuDowngrade
=== CONT  TestAccRedisCache_basic
=== CONT  TestAccRedisCache_PublicNetworkAccessDisabledEnabled
=== CONT  TestAccRedisCache_premiumShardedScaling
=== CONT  TestAccRedisCache_requiresImport
=== CONT  TestAccRedisCache_withoutSSL
--- PASS: TestAccRedisCache_premium (1286.19s)
=== CONT  TestAccRedisCache_ReplicasPerPrimary
--- PASS: TestAccRedisCache_premiumSharded (1424.96s)
=== CONT  TestAccRedisCache_SkuDowngrade
--- PASS: TestAccRedisCache_basic (1787.79s)
=== CONT  TestAccRedisCache_identity
--- PASS: TestAccRedisCache_withoutSSL (1924.74s)
=== CONT  TestAccRedisCache_redisConfiguration
--- PASS: TestAccRedisCache_requiresImport (2086.36s)
=== CONT  TestAccRedisCache_TenantSettings
--- PASS: TestAccRedisCache_PublicNetworkAccessDisabledEnabled (2116.64s)
=== CONT  TestAccRedisCache_RedisVersionUpgrade
--- PASS: TestAccRedisCache_standard (2406.44s)
=== CONT  TestAccRedisCache_RedisVersion
--- PASS: TestAccRedisCache_ReplicasPerPrimary (1233.76s)
=== CONT  TestAccRedisCache_SubscribeAllEvents
=== CONT  TestAccRedisCache_ReplicasPerMaster
--- PASS: TestAccRedisCache_premiumShardedScaling (2801.11s)
--- PASS: TestAccRedisCache_TenantSettings (1138.35s)
=== CONT  TestAccRedisCache_WithoutAuth
--- PASS: TestAccRedisCache_RedisVersion (1154.82s)
=== CONT  TestAccRedisCache_InternalSubnetStaticIP
--- PASS: TestAccRedisCache_redisConfiguration (1663.80s)
=== CONT  TestAccRedisCache_InternalSubnet_withZone
--- PASS: TestAccRedisCache_SubscribeAllEvents (1083.53s)
=== CONT  TestAccRedisCache_InternalSubnet
--- PASS: TestAccRedisCache_identity (1980.89s)
=== CONT  TestAccRedisCache_AOFBackupEnabledDisabled
--- PASS: TestAccRedisCache_ReplicasPerMaster (1031.51s)
=== CONT  TestAccRedisCache_PatchScheduleUpdated
--- PASS: TestAccRedisCache_RedisVersionUpgrade (1726.83s)
=== CONT  TestAccRedisCache_PatchSchedule
--- PASS: TestAccRedisCache_PatchSchedule (1170.31s)
=== CONT  TestAccRedisCache_BackupEnabledDisabled
--- PASS: TestAccRedisCache_PatchScheduleUpdated (1266.67s)
=== CONT  TestAccRedisCache_AOFBackupEnabled
--- PASS: TestAccRedisCache_AOFBackupEnabledDisabled (2006.28s)
=== CONT  TestAccRedisCache_BackupEnabled
--- PASS: TestAccRedisCache_AOFBackupEnabled (1226.56s)
=== CONT  TestAccRedisCache_BackupDisabled
--- PASS: TestAccRedisCache_BackupEnabledDisabled (1346.92s)
--- PASS: TestAccRedisCache_SkuDowngrade (5010.05s)
--- PASS: TestAccRedisCache_BackupEnabled (1290.73s)
--- PASS: TestAccRedisCache_BackupDisabled (1154.55s)
=== CONT  TestAccRedisCache_WithoutAuth
--- PASS: TestAccRedisCache_WithoutAuth (4672.42s)
=== CONT  TestAccRedisCache_InternalSubnetStaticIP
--- PASS: TestAccRedisCache_InternalSubnetStaticIP (4346.97s)
=== CONT  TestAccRedisCache_InternalSubnet_withZone
--- PASS: TestAccRedisCache_InternalSubnet_withZone (4332.63s)
=== CONT  TestAccRedisCache_InternalSubnet
--- PASS: TestAccRedisCache_InternalSubnet (4333.60s)
PASS
PASS    github.com/hashicorp/terraform-provider-azurerm/internal/services/redis 7969.977s
PASS
```

**Review by katbyte — CHANGES_REQUESTED** [review]
test failure
image

**Comment by xuzhang3** [comment]
@katbyte  I cannot reproduce this error. Does there any other errors? Can you help retrigger the CI?

**Review by katbyte — CHANGES_REQUESTED** [review]
it seems consistent in our test subscription:
image

**Comment by xuzhang3** [comment]
@katbyte   I tried different location and subs, cannot reproduce this error.  I run multil-times of the AccTest in team city and  the test will pass some times (1/4 or 1/5), not sure if the is related with the sub. Also I checked the service backend logs, no error found.

**Review by katbyte — APPROVED** [review]
Can see that! LGTM now that test pass 🌵 
image

**Pull request merged**